inc/admin.php

```diff
@@ -70,14 +71,16 @@ function nkdeletepending_options_page() {
 					*/
 					global $wpdb;
 					$wpdb->query( 
-						$wpdb->prepare( "DELETE FROM $wpdb->comments WHERE comment_approved = 0" )
+						$wpdb->prepare( "DELETE FROM $wpdb->comments WHERE comment_approved = %d", 0 )
 					);
-					$deleted_comment_ids = wp_list_pluck( $comments, 'comment_ID' );
-					clean_comment_cache( $deleted_comment_ids );
-					foreach ( $deleted_comment_ids as $comment_id ) {
+					clean_comment_cache( $pending_comment_ids );
```

Comment:
We should not iterate through all the comments. This will not work when we have huge number of comments.

Is there a way to do `clean_comment_cache` without retrieving all the comment ids?

Reply:
Actually, we are looping comment ids for wp_set_comment_status action.  It does accept only one id at a time.